x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/action/FrameGroupID.java

```diff
@@ -30,7 +30,10 @@ public static String getBasenameAndParent(String fullPath) {
 
     public static String create(String fileId, Integer addressOrLine, String exeFilename, String sourceFilename, String functionName) {
         if (Strings.isEmpty(functionName)) {
-            return Integer.toString(Objects.hash(fileId, addressOrLine));
+            if (Strings.isEmpty(exeFilename)) {
```

Comment:
From the PR description and the visualization in the initial commit message, I would have expected some different code.
```
This PR ignores the address or line number when grouping frames with unknown function name but known executable name.
```


I'm understanding the code as this:
```
If exeFilename is empty -> hash(FileID & addressOrLine)
If exeFilename is NOT empty -> hash(exeFileName)
```

From the initial commit message, I would have expected something like this:

```
If functionName is empty -> hash(exeFilename)
Else -> hash(FileID & addressOrLine)
```

Reply:
Good point. Amended the commit message to be more appropriate.